### PR TITLE
tenant adjustment for CRUD role

### PR DESCRIPTION
**Pull request by raosan**
## Description

Since implementation of tenant feature, we cannot user CRUD functionality in Role menu. this PR will fix that so we can CRUD role corresponding to the domain of authenticated user (if superadmin hansip is authenticated, he will need to select tenant in header)

## Test plan
as hansip admin:
1. Login as hansip admin (user with admin role in hansip domain if you use default configuration)
2. do CRUD action in role menu (need to select which tenant is being managed in header)

as non hansip admin:
1. Login as non hansip admin
2. do CRUD action in role menu (domain name is default selected corresponding to authenticated user domain)

### CODE REVIEW

File: components/resources/form/group.tsx [hunk 1]
```diff
@@ -140,7 +140,8 @@ const GroupForm = ({
       isEdit ? `/management/group/${group.rec_id}` : "/management/group",
       {
         group_name: group.group_name,
-        description: group.description
+        description: group.description,
+        group_domain: 'bukalapak.com'
```
Comment (armedi) [inline components/resources/form/group.tsx]:
```suggestion
        group_domain: group.group_domain
```

File: data/tenant.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import { createContext, useContext } from "react";
+import { TenantType } from "./use-get-tenants";
 
 export const defaultTenantContext = {
   tenants: [],
   selected: null,
-  updateTenant: (id: string): void => {}
+  updateTenant: (tenant: any): void => {}
```
Comment (armedi) [inline data/tenant.ts]:
```suggestion
  updateTenant: (tenant: TenantType): void => {}
```

File: components/resources/form/group.tsx (hunk 1 above)
Comment (raosan) [reply components/resources/form/group.tsx]: fixed

File: data/tenant.ts (hunk 2 above)
Comment (raosan) [reply data/tenant.ts]: fixed

File: data/use-get-groups.ts [hunk 3]
```diff
@@ -106,7 +107,7 @@ export const useGetGroup = (groupId: string): GetGroupResult => {
     }
   );
   const { data: rolesData, error: rolesError } = useSWR(
-    `/management/roles`,
+    `/management/tenant/${selectedTenant?.rec_id || null}/roles`,
```
Comment (armedi) [inline data/use-get-groups.ts]:
```suggestion
    selectedTenant ? `/management/tenant/${selectedTenant.rec_id}/roles` : null,
```

using `/management/tenant/${selectedTenant?.rec_id || null}/roles` will cause fetching to this url `/management/tenant/null/roles` when no tenant is selected.
I assume that this intended for not fetching when no tenant is selected, so it requires the key for useSWR to be `null`

File: data/user.ts [hunk 4]
```diff
@@ -71,9 +71,17 @@ export const useWhoAmI = () => {
         ...whoamiData.data.groups.map((group) => group.group_domain)
       ];
 
-      const tenants = tenantsData.tenants
-        .filter((tenant) => domains.includes(tenant.domain))
-        .sort((a, b) => Number(a.name < b.name));
+      const hansipItem = tenantsData.tenants.find(
+        (item) => item.domain === process.env.NEXT_PUBLIC_HANSIP_DOMAIN
+      );
+      let tenants = tenantsData.tenants.filter((tenant) => tenant.domain !== 'hansip')
```
Comment (armedi) [inline data/user.ts]:
```suggestion
      let tenants = tenantsData.tenants.filter((tenant) => tenant.domain !== process.env.NEXT_PUBLIC_HANSIP_DOMAIN)
```

**Review by armedi — APPROVED** [review]
lgtm

**Pull request merged**